### PR TITLE
fix: keep window aligned in corner when resizing, keep width height ratio when adjusting, add max and min size

### PR DESCRIPTION
**Pull request by niajane**

### CODE REVIEW

File: pucoti.py [hunk 1]
```diff
@@ -382,6 +384,20 @@ def table(
         return surf
 
 
+def adjust_window_size(window, scale_factor: float):
+    new_width = window.size[0] * scale_factor
+    new_height = new_width/2.75
```
Comment (ddorn) [inline pucoti.py:389]: Is there a specific reason for the 2.75?

**Review by ddorn — COMMENTED** [review]
I like this! It's definitely something I found frustrating multiple times. Thank you for implementing it! 🤩

File: pucoti.py (hunk 1 above)
Comment (niajane) [reply pucoti.py]: oh yeah, it should be based on the provided dimensions not the default. Thanks :)

Comment (ddorn) [reply pucoti.py]: Okay! Then I think it's probably simpler to use `new_height = window.size[1] * scale_factor` for simplicity

**Review by ddorn — APPROVED** [review]
Awesome! 🍰 🥦 🎉

**Pull request merged**